assets/global.js

```diff
@@ -840,7 +840,11 @@ class VariantSelects extends HTMLElement {
         const html = new DOMParser().parseFromString(responseText, 'text/html')
         const destination = document.getElementById(`price-${this.dataset.section}`);
         const source = html.getElementById(`price-${this.dataset.originalSection ? this.dataset.originalSection : this.dataset.section}`);
+        const variantPickerDestination = document.querySelector('variant-radios') || document.querySelector('variant-selects');
+        const variantPickerSource = html.querySelector('variant-radios') || html.querySelector('variant-selects');
+
         if (source && destination) destination.innerHTML = source.innerHTML;
+        if (variantPickerSource && variantPickerDestination) variantPickerDestination.innerHTML = variantPickerSource.innerHTML;
```

Comment:
While this idea of using the fetched components to avoid using JavaScript to calculate available products is great, right now the main issue with this approach is that we're replacing the variant picker (currently selected element) with something else.

This results in focus being lost every time you interact with the component. Try using a keyboard to navigate between the variant pills - each time it reloads, the focus is lost.

I don't have any alternatives in mind, but as is, this breaks keyboard user experience and causes accessibility issues. Video example below.

https://user-images.githubusercontent.com/37168033/176344283-43d46db9-f67d-4b13-87fe-60866f863d0a.mp4